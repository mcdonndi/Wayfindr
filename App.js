/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import {
    TouchableHighlight,
    StyleSheet,
    Text,
    StatusBar,
    TextInput,
    View
} from 'react-native';
import MapView from 'react-native-maps';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {GooglePlacesAutocomplete} from 'react-native-google-places-autocomplete';

mapStyle = [
    {
        "elementType": "geometry",
        "stylers": [
            {
                "color": "#1d2c4d"
            }
        ]
    },
    {
        "elementType": "labels.text.fill",
        "stylers": [
            {
                "color": "#8ec3b9"
            }
        ]
    },
    {
        "elementType": "labels.text.stroke",
        "stylers": [
            {
                "color": "#1a3646"
            }
        ]
    },
    {
        "featureType": "administrative.country",
        "elementType": "geometry.stroke",
        "stylers": [
            {
                "color": "#4b6878"
            }
        ]
    },
    {
        "featureType": "administrative.land_parcel",
        "elementType": "labels.text.fill",
        "stylers": [
            {
                "color": "#64779e"
            }
        ]
    },
    {
        "featureType": "administrative.province",
        "elementType": "geometry.stroke",
        "stylers": [
            {
                "color": "#4b6878"
            }
        ]
    },
    {
        "featureType": "landscape.man_made",
        "elementType": "geometry.stroke",
        "stylers": [
            {
                "color": "#334e87"
            }
        ]
    },
    {
        "featureType": "landscape.natural",
        "elementType": "geometry",
        "stylers": [
            {
                "color": "#023e58"
            }
        ]
    },
    {
        "featureType": "poi",
        "elementType": "geometry",
        "stylers": [
            {
                "color": "#283d6a"
            }
        ]
    },
    {
        "featureType": "poi",
        "elementType": "labels.text.fill",
        "stylers": [
            {
                "color": "#6f9ba5"
            }
        ]
    },
    {
        "featureType": "poi",
        "elementType": "labels.text.stroke",
        "stylers": [
            {
                "color": "#1d2c4d"
            }
        ]
    },
    {
        "featureType": "poi.park",
        "elementType": "geometry.fill",
        "stylers": [
            {
                "color": "#023e58"
            }
        ]
    },
    {
        "featureType": "poi.park",
        "elementType": "labels.text.fill",
        "stylers": [
            {
                "color": "#3C7680"
            }
        ]
    },
    {
        "featureType": "road",
        "elementType": "geometry",
        "stylers": [
            {
                "color": "#304a7d"
            }
        ]
    },
    {
        "featureType": "road",
        "elementType": "labels.text.fill",
        "stylers": [
            {
                "color": "#98a5be"
            }
        ]
    },
    {
        "featureType": "road",
        "elementType": "labels.text.stroke",
        "stylers": [
            {
                "color": "#1d2c4d"
            }
        ]
    },
    {
        "featureType": "road.highway",
        "elementType": "geometry",
        "stylers": [
            {
                "color": "#2c6675"
            }
        ]
    },
    {
        "featureType": "road.highway",
        "elementType": "geometry.stroke",
        "stylers": [
            {
                "color": "#255763"
            }
        ]
    },
    {
        "featureType": "road.highway",
        "elementType": "labels.text.fill",
        "stylers": [
            {
                "color": "#b0d5ce"
            }
        ]
    },
    {
        "featureType": "road.highway",
        "elementType": "labels.text.stroke",
        "stylers": [
            {
                "color": "#023e58"
            }
        ]
    },
    {
        "featureType": "transit",
        "elementType": "labels.text.fill",
        "stylers": [
            {
                "color": "#98a5be"
            }
        ]
    },
    {
        "featureType": "transit",
        "elementType": "labels.text.stroke",
        "stylers": [
            {
                "color": "#1d2c4d"
            }
        ]
    },
    {
        "featureType": "transit.line",
        "elementType": "geometry.fill",
        "stylers": [
            {
                "color": "#283d6a"
            }
        ]
    },
    {
        "featureType": "transit.station",
        "elementType": "geometry",
        "stylers": [
            {
                "color": "#3a4762"
            }
        ]
    },
    {
        "featureType": "water",
        "elementType": "geometry",
        "stylers": [
            {
                "color": "#0e1626"
            }
        ]
    },
    {
        "featureType": "water",
        "elementType": "labels.text.fill",
        "stylers": [
            {
                "color": "#4e6d70"
            }
        ]
    }
];

export default class App extends Component<{}> {

    constructor (props) {
        super(props);

        this.state = {
            region: {
                latitude: 53.3421508,
                longitude: -6.2535567,
                latitudeDelta: 0.0922,
                longitudeDelta: 0.0421,
            },
            initialRegion: {
                latitude: 53.3421508,
                longitude: -6.2535567,
                latitudeDelta: 0.0922,
                longitudeDelta: 0.0421,
            },
            searchDestinationVis: false
        };
    }

    onRegionChange(region) {
        this.setState({ region });
    }

    render() {
        return (
            <View style={styles.container}>
                <StatusBar
                    backgroundColor="#3F51B5"
                    barStyle="light-content"
                />
                <MapView
                    style={styles.map}
                    initialRegion={this.state.initialRegion}
                    region={this.state.region}
                    onRegionChange={this.onRegionChange.bind(this)}
                    customMapStyle={mapStyle}
                    zoomEnabled={true}
                    scrollEnabled={true}
                />
                <View style={styles.locationSearchView}>
                    <View style={styles.locationSearchViewTop}>
                        <TouchableHighlight style={styles.searchBarButtons}>
                            <Icon
                                style={styles.searchBarIcons}
                                name="menu"
                            />
                        </TouchableHighlight>
                        <GooglePlacesAutocomplete
                            style={styles.locationSearch}
                            placeholder='Enter Location'
                            minLength={2} // minimum length of text to search
                            autoFocus={false}
                            listViewDisplayed='auto'    // true/false/undefined
                            fetchDetails={true}
                            onPress={(data, details = null) => { // 'details' is provided when fetchDetails = true
                                console.log(data, details);
                                this.setState ({region: {
                                    latitude: details.geometry.location.lat,
                                    longitude: details.geometry.location.lng,
                                    latitudeDelta: 0.0922,
                                    longitudeDelta: 0.0421,
                                },
                                searchDestinationVis: true});
                            }}


                            getDefaultValue={() => ''}

                            query={{
                                // available options: https://developers.google.com/places/web-service/autocomplete
                                key: 'AIzaSyCOPygpIBgzbsKYbr1q0Yqc7rvPv6bnhv0',
                                language: 'en', // language of the results
                            }}

                            styles={{
                                container: {
                                    flex: 7
                                },
                                textInputContainer: {
                                    width: '100%'
                                },
                                description: {
                                    fontWeight: 'bold'
                                }
                            }}
                            nearbyPlacesAPI='GooglePlacesSearch' // Which API to use: GoogleReverseGeocoding or GooglePlacesSearch
                            GoogleReverseGeocodingQuery={{
                                // available options for GoogleReverseGeocoding API : https://developers.google.com/maps/documentation/geocoding/intro
                            }}
                            GooglePlacesSearchQuery={{
                                // available options for GooglePlacesSearch API : https://developers.google.com/places/web-service/search
                                rankby: 'distance',
                                types: 'address'
                            }}

                            filterReverseGeocodingByTypes={['locality', 'administrative_area_level_3']} // filter the reverse geocoding results by types - ['locality', 'administrative_area_level_3'] if you want to display only cities


                            debounce={200} // debounce the requests in ms. Set to 0 to remove debounce. By default 0ms.
                        />
                        <TouchableHighlight style={styles.searchBarButtons}>
                            <Icon
                                style={styles.searchBarIcons}
                                name="search"
                            />
                        </TouchableHighlight>
                    </View>
                    <GooglePlacesAutocomplete
                        style={styles.locationSearch}
                        placeholder='Enter Destination'
                        minLength={2} // minimum length of text to search
                        autoFocus={false}
                        listViewDisplayed='auto'    // true/false/undefined
                        fetchDetails={true}
                        onPress={(data, details = null) => { // 'details' is provided when fetchDetails = true
                            console.log(data, details);
                            this.setState ({region: {
                                latitude: details.geometry.location.lat,
                                longitude: details.geometry.location.lng,
                                latitudeDelta: 0.0922,
                                longitudeDelta: 0.0421,
                            }})
                        }}


                        getDefaultValue={() => ''}

                        query={{
                            // available options: https://developers.google.com/places/web-service/autocomplete
                            key: 'AIzaSyCOPygpIBgzbsKYbr1q0Yqc7rvPv6bnhv0',
                            language: 'en', // language of the results
                        }}

                        styles={{
                            container: (!this.state.searchDestinationVis && styles.displayNone) || (styles.displayFlex),
                            textInputContainer: {
                                width: '70%'
                            },
                            description: {
                                width: '70%',
                                fontWeight: 'bold'
                            }
                        }}
                        nearbyPlacesAPI='GooglePlacesSearch' // Which API to use: GoogleReverseGeocoding or GooglePlacesSearch
                        GoogleReverseGeocodingQuery={{
                            // available options for GoogleReverseGeocoding API : https://developers.google.com/maps/documentation/geocoding/intro
                        }}
                        GooglePlacesSearchQuery={{
                            // available options for GooglePlacesSearch API : https://developers.google.com/places/web-service/search
                            rankby: 'distance',
                            types: 'address'
                        }}

                        filterReverseGeocodingByTypes={['locality', 'administrative_area_level_3']} // filter the reverse geocoding results by types - ['locality', 'administrative_area_level_3'] if you want to display only cities


                        debounce={200} // debounce the requests in ms. Set to 0 to remove debounce. By default 0ms.
                    />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        ...StyleSheet.absoluteFillObject,
        justifyContent: 'flex-end',
        alignItems: 'center',
    },
    map: {
        ...StyleSheet.absoluteFillObject,
    },
    locationSearchView: {
        width: '80%',
        flexDirection: 'column',
        alignItems: 'center',
        position: 'absolute',
        top: 20,
    },
    locationSearchViewTop: {
        flex: 1,
        flexDirection: 'row',
        borderRadius: 3,
        backgroundColor: '#E8EAF6',
    },
    displayNone: {
        display: 'none'
    },
    displayFlex: {
        display: 'flex'
    },
    searchBarButtons: {
        flex: 1.5,
    },
    searchBarIcons: {
        ...StyleSheet.absoluteFillObject,
        fontSize: 24,
        color: '#3F51B5',
        textAlignVertical: 'center',
        textAlign: 'center'
    },
    locationSearch: {
        flex: 7,
    }
});
