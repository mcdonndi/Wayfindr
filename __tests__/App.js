import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native-maps', () => {
    const React = require.requireActual('react');
    const MapView = require.requireActual('react-native-maps');

    class MockMapView extends React.Component {
        static Marker = props => React.createElement('Marker', props, props.children);
        static Polyline = props => React.createElement('Polyline', props, props.children);

        render() {
            return React.createElement('MapView', this.props, this.props.children);
        }
    }

    MockMapView.propTypes = MapView.propTypes;
    return MockMapView;
});

it('renders correctly', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('calculateDeltas Function', () => {
    it('returns the correct value', () => {
        let app = new App();
        let expectedValue = app.calculateDelta(4, 2);
        expect(expectedValue).toBe(2);
    })
});
