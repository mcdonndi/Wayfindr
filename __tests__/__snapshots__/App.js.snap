// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "flex-end",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <MapView
    customMapStyle={
      Array [
        Object {
          "elementType": "geometry",
          "stylers": Array [
            Object {
              "color": "#1d2c4d",
            },
          ],
        },
        Object {
          "elementType": "labels.text.fill",
          "stylers": Array [
            Object {
              "color": "#8ec3b9",
            },
          ],
        },
        Object {
          "elementType": "labels.text.stroke",
          "stylers": Array [
            Object {
              "color": "#1a3646",
            },
          ],
        },
        Object {
          "elementType": "geometry.stroke",
          "featureType": "administrative.country",
          "stylers": Array [
            Object {
              "color": "#4b6878",
            },
          ],
        },
        Object {
          "elementType": "labels.text.fill",
          "featureType": "administrative.land_parcel",
          "stylers": Array [
            Object {
              "color": "#64779e",
            },
          ],
        },
        Object {
          "elementType": "geometry.stroke",
          "featureType": "administrative.province",
          "stylers": Array [
            Object {
              "color": "#4b6878",
            },
          ],
        },
        Object {
          "elementType": "geometry.stroke",
          "featureType": "landscape.man_made",
          "stylers": Array [
            Object {
              "color": "#334e87",
            },
          ],
        },
        Object {
          "elementType": "geometry",
          "featureType": "landscape.natural",
          "stylers": Array [
            Object {
              "color": "#023e58",
            },
          ],
        },
        Object {
          "elementType": "geometry",
          "featureType": "poi",
          "stylers": Array [
            Object {
              "color": "#283d6a",
            },
          ],
        },
        Object {
          "elementType": "labels.text.fill",
          "featureType": "poi",
          "stylers": Array [
            Object {
              "color": "#6f9ba5",
            },
          ],
        },
        Object {
          "elementType": "labels.text.stroke",
          "featureType": "poi",
          "stylers": Array [
            Object {
              "color": "#1d2c4d",
            },
          ],
        },
        Object {
          "elementType": "geometry.fill",
          "featureType": "poi.park",
          "stylers": Array [
            Object {
              "color": "#023e58",
            },
          ],
        },
        Object {
          "elementType": "labels.text.fill",
          "featureType": "poi.park",
          "stylers": Array [
            Object {
              "color": "#3C7680",
            },
          ],
        },
        Object {
          "elementType": "geometry",
          "featureType": "road",
          "stylers": Array [
            Object {
              "color": "#304a7d",
            },
          ],
        },
        Object {
          "elementType": "labels.text.fill",
          "featureType": "road",
          "stylers": Array [
            Object {
              "color": "#98a5be",
            },
          ],
        },
        Object {
          "elementType": "labels.text.stroke",
          "featureType": "road",
          "stylers": Array [
            Object {
              "color": "#1d2c4d",
            },
          ],
        },
        Object {
          "elementType": "geometry",
          "featureType": "road.highway",
          "stylers": Array [
            Object {
              "color": "#2c6675",
            },
          ],
        },
        Object {
          "elementType": "geometry.stroke",
          "featureType": "road.highway",
          "stylers": Array [
            Object {
              "color": "#255763",
            },
          ],
        },
        Object {
          "elementType": "labels.text.fill",
          "featureType": "road.highway",
          "stylers": Array [
            Object {
              "color": "#b0d5ce",
            },
          ],
        },
        Object {
          "elementType": "labels.text.stroke",
          "featureType": "road.highway",
          "stylers": Array [
            Object {
              "color": "#023e58",
            },
          ],
        },
        Object {
          "elementType": "labels.text.fill",
          "featureType": "transit",
          "stylers": Array [
            Object {
              "color": "#98a5be",
            },
          ],
        },
        Object {
          "elementType": "labels.text.stroke",
          "featureType": "transit",
          "stylers": Array [
            Object {
              "color": "#1d2c4d",
            },
          ],
        },
        Object {
          "elementType": "geometry.fill",
          "featureType": "transit.line",
          "stylers": Array [
            Object {
              "color": "#283d6a",
            },
          ],
        },
        Object {
          "elementType": "geometry",
          "featureType": "transit.station",
          "stylers": Array [
            Object {
              "color": "#3a4762",
            },
          ],
        },
        Object {
          "elementType": "geometry",
          "featureType": "water",
          "stylers": Array [
            Object {
              "color": "#0e1626",
            },
          ],
        },
        Object {
          "elementType": "labels.text.fill",
          "featureType": "water",
          "stylers": Array [
            Object {
              "color": "#4e6d70",
            },
          ],
        },
      ]
    }
    initialRegion={
      Object {
        "latitude": 53.3421508,
        "latitudeDelta": 0.0922,
        "longitude": -6.2535567,
        "longitudeDelta": 0.0421,
      }
    }
    onRegionChange={[Function]}
    region={
      Object {
        "latitude": 53.3421508,
        "latitudeDelta": 0.0922,
        "longitude": -6.2535567,
        "longitudeDelta": 0.0421,
      }
    }
    scrollEnabled={true}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    zoomEnabled={true}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "position": "absolute",
        "top": 20,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#E8EAF6",
          "borderRadius": 3,
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1.5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "bottom": 0,
                "color": "#3F51B5",
                "fontSize": 24,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "textAlign": "center",
                "textAlignVertical": "center",
                "top": 0,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          î—’
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "flex": 7,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#C9C9CE",
                "borderBottomColor": "#b5b5b5",
                "borderBottomWidth": 0.5,
                "borderTopColor": "#7e7e7e",
                "borderTopWidth": 0.5,
                "flexDirection": "row",
                "height": 44,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            autoFocus={false}
            clearButtonMode="while-editing"
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Enter Location"
            placeholderTextColor="#A8A8A8"
            returnKeyType="default"
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 5,
                  "flex": 1,
                  "fontSize": 15,
                  "height": 28,
                  "marginLeft": 8,
                  "marginRight": 8,
                  "marginTop": 7.5,
                  "paddingBottom": 4.5,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 4.5,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1.5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "bottom": 0,
                "color": "#3F51B5",
                "fontSize": 24,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "textAlign": "center",
                "textAlignVertical": "center",
                "top": 0,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          î¢¶
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "display": "none",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#C9C9CE",
              "borderBottomColor": "#b5b5b5",
              "borderBottomWidth": 0.5,
              "borderTopColor": "#7e7e7e",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "height": 44,
            },
            Object {
              "width": "70%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoFocus={false}
          clearButtonMode="while-editing"
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Enter Destination"
          placeholderTextColor="#A8A8A8"
          returnKeyType="default"
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 5,
                "flex": 1,
                "fontSize": 15,
                "height": 28,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 7.5,
                "paddingBottom": 4.5,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 4.5,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
</View>
`;
